mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -414,6 +417,15 @@ public MailboxSession login(String userid, String passwd, Logger log) throws Bad
         }
     }
 
+    @Override
+    public MailboxSession loginAsOtherUser(String adminUserid, String passwd, String realUserId, Logger log) throws BadCredentialsException, MailboxException {
+        if (login(adminUserid, passwd) && authorizator.canLoginAsOtherUser(adminUserid, realUserId)) {
+            return createSystemSession(realUserId, log);
+        } else {
+            throw new BadCredentialsException();
```

Comment:
nope, generally we try to not disclose too much information when authenticating or authorizing people